### DISCUSSION TITLE
How can I create protected routes?

### DISCUSSION THREAD

**Discussion by RWallan** [reactive-python/reactpy-router#28] [Question] [answered]
Hi,

I'm learning reactpy and reactpy-routers with a simple to-do app. In this app, I'll have users that can see only your tasks and only authenticated users can see tasks.

I already have a backend developed with FastAPI with a login endpoint that returns the access_token. So how can I protect my routes to allow only authenticated users?

**Comment by Archmonger** [comment, marked as answer]
Routes can be protected within the individual components that render those routes.

For example,

```python
from reactpy import component, html

@component
def example():
    if condition_is_met:
       # You can either redirect via scripts or just return None.
       return html.script('window.location.href = "/my_url/";')

    return ... # Your ReactPy application
```

**Reply by RWallan** [reply]
Something that I can use is maybe temp files? To work like an local storage from js and store the access token (or other infos from user)? So with this I can load the info and check the condition peharps?
